private/pkg/normalpath/normalpath_unix_test.go

```diff
@@ -319,6 +319,23 @@ func TestEqualsOrContainsPath(t *testing.T) {
 	testEqualsOrContainsPath(t, true, "b", "b/a/c")
 }
 
+func BenchmarkEqualsOrContainsPath(b *testing.B) {
+	b.Run("Short", func(b *testing.B) {
+		var ok bool
+		for i := 0; i < b.N; i++ {
+			ok = EqualsOrContainsPath("a/b/c", "a/b/c/d", Relative)
+		}
+		assert.True(b, ok)
+	})
+	b.Run("Long", func(b *testing.B) {
+		var ok bool
+		for i := 0; i < b.N; i++ {
+			ok = EqualsOrContainsPath("a/b/c", "a/b/c/d/e/f/g/h/i/j", Relative)
+		}
+		assert.True(b, ok)
+	})
+}
+
```

Comment:
+       pathRoot := "/"
+       if pathType == Relative {
+               pathRoot = "."
        }
-       return false
+       // If the value is the root, it contains everything.
+       return value == pathRoot ||
+               // If the value is the path, it contains itself.
+               strings.EqualFold(value, path) ||
+               // If the value is a directory and the path is in the directory.
+               // Check has prefix and the next character is a path separator.
+               (len(path) > len(value) &&
+                       strings.EqualFold(path[:len(value)], value) &&
+                       (value[len(value)-1] == '/' || path[len(value)] == '/'))